### PR TITLE
GT-1448 Complete GetToolTranslationsUseCase for downloading tool language translations

### PR DESCRIPTION
**Pull request by levieggertcru**

### CODE REVIEW

**Comment by levieggertcru** [comment]
Hey @rachaelblue this is another pretty large PR.

- Cleaned up TranslationsRepository.  Also realized the GodToolsToolParser Manifest can't be shared between threads.  I think this is because of the watchers. 
- Added LanguageSettingsRepository which will replace LanguageSettingsService.
- Added UseCases for working with the language settings repository. 
- Completed GetToolTranslationsUseCase for downloading tool language translations.
- Added downloading of attachment files to AttachmentsRepository.

File: godtools/App/Share/Data/LanguageSettingsRepository/Cache/LanguageSettingsCache.swift
Comment (rachaelblue) [inline godtools/App/Share/Data/LanguageSettingsRepository/Cache/LanguageSettingsCache.swift:69]: Nice to have these in UserDefaults!

File: godtools/App/Share/Data/TranslationsRepository/Api/MobileContentTranslationsApi.swift
Comment (rachaelblue) [inline godtools/App/Share/Data/TranslationsRepository/Api/MobileContentTranslationsApi.swift:42]: Haven't seen `tryMap` before, that's cool!

File: godtools/App/Share/Data/TranslationsRepository/TranslationsRepository.swift
Comment (rachaelblue) [inline godtools/App/Share/Data/TranslationsRepository/TranslationsRepository.swift:79]: Does `collect` wait until all the publishers in the merge have a response, and then notify subscribers only once with all the values?

File: godtools/App/Share/Domain/UseCases/GetDeviceLanguageUseCase/GetDeviceLanguageUseCase.swift
Comment (rachaelblue) [inline godtools/App/Share/Domain/UseCases/GetDeviceLanguageUseCase/GetDeviceLanguageUseCase.swift:13]: `GetDeviceLanguageUseCase` doesn't seem to be using `GetLanguageUseCase`.  Should it?

File: godtools/App/Features/SetupParallel/Views/ParallelLanguageListViewModel.swift
Comment (levieggertcru) [reply godtools/App/Features/SetupParallel/Views/ParallelLanguageListViewModel.swift:39]: Hey @rachaelblue I had to double check this one, found these marble diagrams that do a good job explaining the difference.
https://rxmarbles.com/#zip
https://rxmarbles.com/#combineLatest

File: godtools/App/Share/Data/TranslationsRepository/TranslationsRepository.swift
Comment (levieggertcru) [reply godtools/App/Share/Data/TranslationsRepository/TranslationsRepository.swift:79]: Hey @rachaelblue from my understanding the collect will gather all of the values of the upstream publishers once those have all completed and completed successfully.  It seems if any of the upstream publishers fail, then the entire thing immediately fails.

https://developer.apple.com/documentation/combine/publishers/mergemany/collect()

File: godtools/App/Share/Domain/UseCases/GetDeviceLanguageUseCase/GetDeviceLanguageUseCase.swift
Comment (levieggertcru) [reply godtools/App/Share/Domain/UseCases/GetDeviceLanguageUseCase/GetDeviceLanguageUseCase.swift:13]: I was contemplating that as well.  There are some differences between them though.
- The LanguageDomainModel is mapped from mobile content api languages with applied business logic.
- Where as DeviceLanguageDomainModel is mapped from the OS System Device languages, in which we aren't applying any special business logic at this time.  We mostly use this for enabling specific device language features in the app.  

There is a relationship between the 2 of these where the language codes from our languages in mobile content api match a Locale identifier.

For now, let's just keep them distinct.  This way we can grab the current device language without depending on the mobile content languages api.

File: godtools/App/Features/SetupParallel/Views/ParallelLanguageListViewModel.swift
Comment (frett) [reply godtools/App/Features/SetupParallel/Views/ParallelLanguageListViewModel.swift:39]: they are standard reactive stream operators, `zip` will take elements in pairs. `combineLatest` will combine the latest value from both streams whenever either stream emits a value.

`Zip(['a', 'b', 'c'], [0, 1])` will process the pairs `['a',0], ['b',1]`. if the second stream never produces a 3rd element, then `c` will never be processed from the first stream.

`combineLatest` will always trigger when a new value is received.

the marbles diagrams are great at visualizing this.

Comment (frett) [reply godtools/App/Features/SetupParallel/Views/ParallelLanguageListViewModel.swift:39]: I've actually not found a usage for zip anywhere in Android yet, usually I'm looking for `combineLatest` to transform 2 streams into a single stream

Comment (levieggertcru) [reply godtools/App/Features/SetupParallel/Views/ParallelLanguageListViewModel.swift:39]: Not sure if Zip really makes much sense here now.  Seems I would always want the latest of these.

Comment (levieggertcru) [reply godtools/App/Features/SetupParallel/Views/ParallelLanguageListViewModel.swift:39]: Thanks for the added input Daniel!  Going to update to use latest.

File: godtools/App/Features/Languages/ChooseLanguage/ChooseLanguageViewModel.swift
Comment (rachaelblue) [inline godtools/App/Features/Languages/ChooseLanguage/ChooseLanguageViewModel.swift:71]: Does this need to be `CombineLatest` then, too?  Seems like it should be. Can you combine more than 2 streams?

File: godtools/App/Share/Domain/UseCases/GetDeviceLanguageUseCase/GetDeviceLanguageUseCase.swift
Comment (rachaelblue) [reply godtools/App/Share/Domain/UseCases/GetDeviceLanguageUseCase/GetDeviceLanguageUseCase.swift:13]: Makes sense to me.  Can we remove this useCase dependency then?

File: godtools/App/Features/Languages/ChooseLanguage/ChooseLanguageViewModel.swift
Comment (levieggertcru) [reply godtools/App/Features/Languages/ChooseLanguage/ChooseLanguageViewModel.swift:71]: Hey @rachaelblue this should also use CombineLatest.   I will include this in GT-1746

**Review by rachaelblue — APPROVED** [review]

**Pull request merged**
